core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -278,28 +344,38 @@ public void close() throws IOException {
 
         @Override
         public CloseableIterator<ScanTask> iterator() {
+          // Partition filter by base table filter
           Expression partitionFilter =
               Projections.inclusive(spec, isCaseSensitive()).project(baseTableFilter);
 
-          // Filter partitions
+          // Read manifests (use original table's partition ids to de-serialize partition values)
           CloseableIterable<ManifestEntry<DeleteFile>> deleteFileEntries =
-              ManifestFiles.readDeleteManifest(manifest, table().io(), transformedSpecs)
+              ManifestFiles.readDeleteManifest(manifest, table().io(), table().specs())
```

Comment:
Here is a bit tricky.  We need to read manifests using the original field ids, so go back to use the original table's partition spec which preserves the original field ids.  

We can no longer use the spec to bind the user-provided partition filter though, because now those field ids are the old ones and do not match the reasisgned ones, so we need to split that out and evaluate it separately outside the ManifestReader.